### DISCUSSION TITLE
Problem with download a quite simple web-page - server returns different content

### DISCUSSION THREAD

**Discussion by cscode2000** [curl/curl#12334] [Q&A] [answered]
I really tried a lot but can't get the same HTML-content with curl like with Torbrowser (Firefox).
After loggin the session with Network manager I set all headers to the same value in curlrc, but get only a WAF Challenge / Bot protection page back. With TB/FF the right content is shown (weekly market advertisements). What am I doing wrong? **Javascript is disabled!**

curl.exe --cacert .\curl-ca-bundle.crt -K .\curlrc.txt -o outfile https://www.rewe.de/angebote/ginsheim-gustavsburg/1463902/rewe-markt-darmstaedter-landstrasse-66-68/

curlrc.txt:
-----------------
http2
cookie = cookies
cookie-jar = cookies
socks5-hostname = 127.0.0.1:9150
user-agent = "Mozilla/5.0 (Windows NT 10.0; rv:102.0) Gecko/20100101 Firefox/102.0"
header = "Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,*/*;q=0.8"
header = "Accept-Language: en-US,en;q=0.5"
header = "Accept-Encoding: gzip, deflate, br"
header = "Connection: keep-alive"
header = "Upgrade-Insecure-Requests: 1"
header = "Sec-Fetch-Dest: document"
header = "Sec-Fetch-Mode: navigate"
header = "Sec-Fetch-Site: none"
header = "Sec-Fetch-User: ?1"

**Comment by vszakats** [comment, marked as answer]
This issue isn't specific to curl-for-win builds of curl.

You may be interested in these projects, which make curl impersonate browsers:
https://github.com/lwthiker/curl-impersonate
https://github.com/depler/curl-impersonate-win

Also see this existing thread: #11192

**Comment by cscode2000** [comment]
> You may be interested in these projects, which make curl impersonate browsers: 

Thank you, I already had the hunch, that this might be the reason. MSYS2, Pacman, patch... That's too much additional sources for just one crappy shitty web site (here and now) probably I use AutoHotkey with browser. Maybe sometimes it will be implemented in the regular curl, when others went into this problem, too.
